torch/csrc/jit/mobile/flatbuffer_loader.cpp

```diff
@@ -296,6 +296,11 @@ mobile::Module FlatbufferLoader::parseModule(
       "Parsing flatbuffer module: Corrupted ivalues/object_types field");
   TORCH_CHECK(
       reinterpret_cast<const char*>(ivalues) < end, "Corrupted ivalues field");
+  TORCH_CHECK(
+      ivalues->size() >= 0 && module->object_types()->size() >= 0 &&
```

Comment:
looking at the `mobile::serialization::Module` code:

https://github.com/pytorch/pytorch/blob/12d6446507df794e5f1f563250bbbd8bbd08044b/torch/csrc/jit/serialization/mobile_bytecode_generated.h#L2265

it seems like `ivalues` and `module->object_types()` are both `flatbuffers::Vector`, which use an unsigned return type

https://flatbuffers.dev/classflatbuffers_1_1_vector.html

(reference to `uoffset_t` being unsigned: https://flatbuffers.dev/md__internals.html)

should we just check the storage_data_size() instead, since that's the only part that can be signed? Or is there another factor here that can result in the other size() fields returning signed values?

Reply:
Yeah, it seems like only `module->storage_data_size()` should be checked here as int32_t.